ShinyExpenseTracker/src/ca/ualberta/cs/shinyexpensetracker/TagController.java

```diff
@@ -1,21 +1,78 @@
 package ca.ualberta.cs.shinyexpensetracker;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 import ca.ualberta.cs.shinyexpensetracker.models.TagList;
 
+/**
+ * TagController object that stores a list
+ * of created tags. It is unique for the
+ * program
+ * @author Oleg Oleynikov
+ * @version 1.0
+ * @since 2015-03-08
+ */
+
 public class TagController {
-	private TagList list;
+	private static TagList tagList;
 	
+	/**
+	 * Returns the current list of tags. If there is none,
+	 * returns a new blank tag list.
+	 * <p>
+	 * @return the current tag list
+	 */
+	static public TagList getTagList()
+	{
+		if (tagList == null)
+		{
+			tagList = new TagList();
+		}
+		return tagList;
+	}
+	
+	// I don't think we need these if it is a singleton.
+	/*
 	public TagController() {
-		list = new TagList();
+		tagList = new TagList();
 	}
+	
 	public TagController(TagList list) {
-		this.list = list;
+		this.tagList = list;
 	}
+	*/
 	
-	public void addTag(String s) {
-		list.addTag(s);
+
+	/**
+	 * Add a tag to the current tag list
+	 * <p>
+	 * @param s the string value of the added tag
+	 */
+	public static void addTag(String s) {
+		Pattern p = Pattern.compile("^[a-zA-Z0-9]*$");
+		Matcher m = p.matcher(s);
+		if (m.matches())
+			tagList.addTag(s);
+		else
+			return;
+	}
```

Comment:
* It now displays a Toast and asks for a new input
* Not anymore
* Isn't that kind of handled by ArrayList?